File: Example/ObjectiveC Example/Channel Plugin Sample objc/MainViewController.m
//
//  MainViewController.m
//  Channel Plugin Sample objc
//
//  Created by Haeun Chung on 21/03/2017.
//  Copyright © 2017 ZOYI. All rights reserved.
//

#import "MainViewController.h"
#import <CHPlugin/CHPlugin-Swift.h>

@interface MainViewController ()

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
  
  if (!self.loaded) {
    self.isUser ? [self loginAsUser] : [self loginAsVeil];
    self.loaded = YES;
  }
}

- (void)viewWillDisappear:(BOOL)animated
{
  [super viewWillDisappear:animated];
  
  if (self.isMovingFromParentViewController) {
    [ChannelPlugin checkOut];
  }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)loginAsUser {
  if ([self.userId isEqualToString:@""]) {
    return;
  }
  
  CheckIn *checkin = [[CheckIn alloc] init];
  [checkin withName:@""];
  [checkin withUserId:self.userId];
  [checkin withMobileNumber:@""];
  
  [ChannelPlugin checkIn:checkin completion:^(enum ChannelCheckInCompletionStatus state) {
    switch (state) {
      case ChannelCheckInCompletionStatusSuccess:
        break;
      default:
        self.loginLabel.text = @"Login failed due to invalid parameters";
        self.detailLabel.text = @"Please go back and try again";
    }
  }];
}
  
- (void)loginAsVeil {
  [ChannelPlugin checkIn:nil completion:^(enum ChannelCheckInCompletionStatus state) {
    
  }];
}

@end

File: Example/Swift Example/Channel Plugin Sample/MainViewController.swift
//
//  MainViewController.swift
//  Example
//
//  Created by Haeun Chung on 14/03/2017.
//  Copyright © 2017 ZOYI. All rights reserved.
//

import Foundation
import UIKit
import ChannelIO

class MainViewController : UIViewController, ChannelIODelegate {
  @IBOutlet var loginTypeLabel: UILabel!
  @IBOutlet var descLabel: UILabel!
  
  var loaded: Bool = false
  var isUser: Bool = false
  var userId: String = ""
  var userName: String = ""
  var phoneNumber: String = ""
  override func viewDidLoad() {
    super.viewDidLoad()
    ChannelIO.delegate = self
    self.loginTypeLabel.text = self.isUser
      ? "Checked in (user)" : "Checked in (veil)"
    self.descLabel.text = "Check out when navigates back"
  }
  
  override func viewWillAppear(_ animated: Bool) {
    super.viewWillAppear(animated)
    
    if self.loaded == false {
      self.isUser ? self.loginAsUser() : self.loginAsVeil()
      self.loaded = true
    }
  }
  
  override func viewWillDisappear(_ animated: Bool) {
    super.viewWillDisappear(animated)
    
    if self.isMovingFromParentViewController {
      ChannelIO.shutdown()
    }
  }
  
  func willShowChatList() {
    print("will show")
  }
  
  func willHideChatList() {
    print("will hide")
  }
  
  func onClickChatLink(url: URL) -> Bool {
    return true
  }
  
  func loginAsUser() {
    if self.userId == "" {
      return
    }
    
    let guest = Guest()
      .set(name: self.userName)
      .set(id: self.userId)
      .set(mobileNumber: self.phoneNumber)
    
    let settings = ChannelPluginSettings(pluginKey: "52eb6f27-38c7-476d-ad92-83e6299b7e07")
    
    ChannelIO.boot(with: settings, guest: guest) { (completion) in
      switch completion {
      case .success:
        break
      default:
        self.loginTypeLabel.text = "Login failed due to invalid parameters"
        self.descLabel.text = "Please go back and try again"
      }
    }
  }
  
  func loginAsVeil() {
    let settings = ChannelPluginSettings(pluginKey: "52eb6f27-38c7-476d-ad92-83e6299b7e07")
    
    ChannelIO.boot(with: settings) { completion in
      
    }
  }
}
